src/core/app/features.ts

```diff
@@ -0,0 +1,23 @@
+import { appConfig } from '~core/app_config';
+import { apiClient } from '~core/apiClientInstance';
+import type { AppFeatureType, BackendFeature } from '~core/auth/types';
+
+const effectiveFeatures = {};
+
+// intersection of both, user can use his feature if app has it
+function mergeFeatures(global, custom, user) {
+  const res = { ...global, ...custom, ...user };
+  return res;
+}
+
+export function hasFeature(f: AppFeatureType) {
```

Comment:
it feels ok here, but I remember we described it in the team that 1 letter variable names acceptable only in .map .foreach etc callbacks